tests/fXmlRpc/Serializer/AbstractSerializerTest.php

```diff
@@ -24,35 +24,14 @@
 
 namespace fXmlRpc\Serializer;
 
-use DateTime;
-use DateTimeZone;
-use fXmlRpc\Value\Base64;
 use PHPUnit\Framework\TestCase;
 
 abstract class AbstractSerializerTest extends TestCase
 {
     /** @var SerializerInterface */
     protected $serializer;
 
-    public function provideTypes()
-    {
-        return array(
-            array('string', 'test string', 'test string'),
-            array('int', 2, '2'),
-            array('int', -2, '-2'),
-            array('double', 1.2, '1.2'),
-            array('double', -1.2, '-1.2'),
-            array('boolean', true, '1'),
-            array('boolean', false, '0'),
-            array(
-                'dateTime.iso8601',
-                DateTime::createFromFormat('Y-m-d H:i:s', '1998-07-17 14:08:55', new DateTimeZone('UTC')),
-                '19980717T14:08:55'
-            ),
-            array('base64', Base64::serialize('string'), "c3RyaW5n\n"),
-            array('string', 'Ümläuts', '&#220;ml&#228;uts'),
-        );
```

Comment:
What was the issue with this and PHP 8? Foreign classes no longer supported as data providers?

Reply:
It's more with the phpunit update.

Some of the classes are using `'1.200000'` some `1.2`.
With some old version of phpunit, it was considered as the same value. Now it doesn't.

Reply:
There was a previous discussion about https://github.com/lstrojny/fxmlrpc/pull/81#discussion_r543409485

By moving the data provider I still execute all the same tests.